### DISCUSSION TITLE
Getting 🚫 ope: no songs found when trying to play a Spotify playlist

### DISCUSSION THREAD

**Discussion by bendinwire** [codetheweb/muse#946] [Q&A] [answered]
Hi Everyone,

New to Muse here - have it installed in a Docker container. I've added to my server and am trying this command:
`/play query: https://open.spotify.com/track/716eVtciaJCtTcaknS3QXr?si=8f4ddffb159a4f18`
I'm getting:
`🚫 ope: no songs found`
After running it. I have tried a few other playlists as well. Any suggestions? The log on the docker instance doesn't really show anything helpful that I can see:
```> muse@2.3.0 env:set-database-url
> tsx src/scripts/run-with-database-url.ts

========================================
========================================
================= muse =================
========================================
=========== Made with 🎶 by ============
==== https://github.com/codetheweb =====
========================================
========================================

Running version 2.3.0 built on 5/14/2023 (commit 47a2adb1dd2b9a409b31fdda536574aeac6901d7)

🔧 Need help? https://github.com/codetheweb/muse/discussions/new
🐛 Bug? https://github.com/codetheweb/muse/issues/new
✨ New feature? https://github.com/codetheweb/muse/issues/new
☕ Discussion? https://github.com/codetheweb/muse/discussions/new
💰 Help me? https://www.paypal.me/codetheweb https://github.com/sponsors/codetheweb


- Applying database migrations...
✔ Database migrations applied.
- 📡 connecting to Discord...
✔ Ready! Invite the bot with https://discordapp.com/oauth2/au <snip>
```

Any suggestions?

Thanks

**Comment by codetheweb** [comment]
You can get more logs if you set the `DEBUG` environment variable to `*`.

Do YouTube links work?

**Comment by darkslyde** [comment]
I too have this issue with spotify. I am wondering if i incorrectly setup the spotify account. youtube links work, but selecting a query from youtube gives ope: filterWrapper is not iterable error

**Reply by darkslyde** [reply]
![image](https://github.com/codetheweb/muse/assets/66486427/b088e384-6a25-4955-9664-f9e6b4d33924)

i would get:

+0ms
  prisma:client:libraryEngine  sending request, this.libraryStarted: true +263ms
  muse Cache miss: ["queen don't stop me now",{"limit":10}] +2s
  muse /usr/app/node_modules/ytsr/lib/utils.js:23
  muse   for (const filterGroup of filterWrapper) {
  muse                             ^
  muse 
  muse TypeError: filterWrapper is not iterable
  muse     at Object.exports.parseFilters (/usr/app/node_modules/ytsr/lib/utils.js:23:29)
  muse     at module.exports (/usr/app/node_modules/ytsr/lib/main.js:60:24)
  muse     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
  muse     at KeyValueCacheProvider.wrap (/usr/app/src/services/key-value-cache.ts:53:20)
  muse     at run (/usr/app/node_modules/p-queue/dist/null:109:36) +840ms

![image](https://github.com/codetheweb/muse/assets/66486427/a8e91395-b420-49df-bc75-44b72ecabc74)

and I get

 +0ms
  prisma:client:libraryEngine  sending request, this.libraryStarted: true +435ms
  muse Cache miss: ["\"Don't Stop Me Now\" \"Queen\"",{"limit":10}] +4s
  muse /usr/app/src/services/add-query-to-queue.ts:114
  muse       throw new Error('no songs found');
  muse             ^
  muse 
  muse Error: no songs found
  muse     at AddQueryToQueue.addToQueue (/usr/app/src/services/add-query-to-queue.ts:114:13)
  muse     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
  muse     at play_default.execute (/usr/app/src/commands/play.ts:49:5)
  muse     at Client.<anonymous> (/usr/app/src/bot.ts:79:13) +759ms

**Reply by codetheweb** [reply, marked as answer]
Probably related to https://github.com/codetheweb/muse/issues/948, please upgrade and see if it's fixed.

**Reply by darkslyde** [reply]
Issues resolved as of MUSE version 2.4.2 built on 8/12/2023! Thank you!
Btw, I'm running it as a docker on Unraid 6.12.3 if anyone else is curious.
